### PR TITLE
Add an AUTHORS file and simplify copyright notices.

### PR DESCRIPTION
**Pull request by 23Skidoo**
Instead of having multiple incomplete lists of contributors in various places, list them all in a single file.

### CODE REVIEW

**Comment by phadej** [comment]
Could you add the script used to generate the `CONTRIBUTORS` file. Maybe adding the email addresses would be worth too.

**Comment by 23Skidoo** [comment]
The reason I didn't include e-mail addresses is that you get more duplicates that way,

**Comment by phadej** [comment]
@23Skidoo yes, but if we pick single address per name (e.g. first, or the one pointed by the person), and also single name per address.

**Comment by 23Skidoo** [comment]
Disabled the Travis check (it fails due to Travis using `--depth=50`).

Also moved the generator script (which is just `git shortlog -s | cut -f 2-`) to a separate file.

**Comment by 23Skidoo** [comment]
@phadej I think improvements can be made later, let's start with something simple. Ideally the script should also deal with multiple spellings of the same name (e.g. "Andres Loeh" vs. "Andres Löh").

**Comment by 23Skidoo** [comment]
One other complication is that some commits are not correctly attributed (mostly early ones).

**Comment by hvr** [comment]
@23Skidoo 

> Ideally the script should also deal with multiple spellings of the same name (e.g. "Andres Loeh" vs. "Andres Löh").

If you generate the information from Git, then there's a facility to fixup entries available in Git, see e.g. http://git.haskell.org/ghc.git/blob/HEAD:/.mailmap

**Comment by 23Skidoo** [comment]
@hvr Thanks, will look into it.

**Comment by 23Skidoo** [comment]
Added a `.mailmap` file; added e-mails to the `AUTHORS` file.

File: Cabal/LICENSE [hunk 1]
```diff
@@ -1,8 +1,4 @@
-Copyright (c) 2003-2016, Isaac Jones, Simon Marlow, Martin Sjögren,
-                         Bjorn Bringert, Krasimir Angelov,
-                         Malcolm Wallace, Ross Patterson, Ian Lynagh,
-                         Duncan Coutts, Thomas Schilling,
-                         Johan Tibell, Mikhail Glushenkov
+Copyright (c) 2003-2016, Cabal Development Team
```
Comment (dcoutts) [inline Cabal/LICENSE]: Suggest referring to the authors list, e.g.
```
Copyright (c) 2003-2016, Cabal Development Team 
See the AUTHORS file for the full list of copyright holders
```

File: cabal-install/cabal-install.cabal [hunk 2]
```diff
@@ -9,17 +9,9 @@ homepage:           http://www.haskell.org/cabal/
...
-                    Bjorn Bringert <bjorn@bringert.net>
-                    Isaac Potoczny-Jones <ijones@syntaxpolice.org>
-                    Duncan Coutts <duncan@community.haskell.org>
-Maintainer:         cabal-devel@haskell.org
-Copyright:          2005 Lemmih <lemmih@gmail.com>
-                    2006 Paolo Martini <paolo@nemail.it>
-                    2007 Bjorn Bringert <bjorn@bringert.net>
-                    2007 Isaac Potoczny-Jones <ijones@syntaxpolice.org>
-                    2007-2012 Duncan Coutts <duncan@community.haskell.org>
+Author:             Cabal Development Team
```
Comment (dcoutts) [inline cabal-install/cabal-install.cabal]: Possibly again refer to AUTHORS file e.g.
```
Author:             Cabal Development Team (see AUTHORS file)
```

**Comment by dcoutts** [comment]
@23Skidoo I think I'd prefer to have my primary contributor email listed as the community.haskell.org one. I have used my company address for contributions where those were on company time, but that's the small minority.

**Comment by 23Skidoo** [comment]
@dcoutts All fixed.

**Comment by dcoutts** [comment]
@23Skidoo Thanks!

**Pull request merged**